src/components/FormSelectionContainer/FormSelect/FormSelect.tsx

```diff
@@ -18,27 +18,28 @@ export const FormSelect: FunctionComponent<FormSelectProps> = ({ id, form, onAdd
   const refButton = useRef<HTMLDivElement>(null);
 
   const handleForm = async (): Promise<void> => {
+    if (isValidDns) {
+      onAddForm();
+    } else {
+      ReactTooltip.show(refButton.current as unknown as Element);
+    }
+  };
+
+  const dnsCheck = useCallback(async () => {
     if (config?.network === "local") {
       setIsValidDns(true); // for local e2e to pass, skip dns validate + set valid
     } else {
       const isDnsValidated = await validateDns(form);
       if (!isDnsValidated) {
-        setErrorMsg(
-          `${getIssuerAddress(form.defaults)} address could not be found at ${getIssuerLocation(
-            form.defaults
-          )} DNS TXT records.`
-        );
-        ReactTooltip.show(refButton.current as unknown as Element);
+        setErrorMsg(`The contract address could not be found on ${getIssuerLocation(form.defaults)} DNS TXT records.`);
       }
       setIsValidDns(isDnsValidated);
     }
-  };
+  }, [config, form]);
 
   useEffect(() => {
-    if (isValidDns) {
-      onAddForm();
-    }
-  }, [isValidDns, onAddForm]);
+    dnsCheck();
+  }, [dnsCheck]);
 
```

Comment:
oh think the intention was avoid making unnecessary calls to google dns, for as long as user did not initiate click